src/Datalake2Anomali.py

```diff
@@ -127,6 +137,163 @@ def uploadPayload(self, payload):
+        mapped_blocks = [
+            PatchTipReportModel(
+                body=block['executive_summary'],
+                modified_ts=datetime.strftime(parser.parse(block['timestamp_updated']), ANOMALI_TIME_FORMAT),
+                name=block["title"],
+                tags=[
+                    self.GENERIC_WORLD_WATCH_BULLETIN_TAG,
+                    self.get_world_watch_tag(tipreport_id),
+                    *block['tags']
+                ]
+            ).model_dump()
+            for block in content_block_list
+        ]
+
+        compared_dt = datetime.now() - timedelta(hours=config.upload_frequency)
+        if last_modified:
+            compared_dt = last_modified
+
+        mapped_blocks = list(
+            filter(
+                lambda x: parser.parse(x['modified_ts']).replace(tzinfo=timezone.utc) > compared_dt.replace(tzinfo=timezone.utc),
+                mapped_blocks
+            )
+        )
+
+        return mapped_blocks
+
+
+    def patch_existing_tipreport(self, advisory, tipreport_id, last_modified):
+        patch_url = f"{self.anomali_url}/api/v1/tipreport/{tipreport_id}/"
+
+
+        mapped_blocks = self.map_ww_content_block_to_anomali(tipreport_id, advisory['content_blocks'], last_modified)
+
+        patch_payload = {
+            "objects": mapped_blocks
+        }
+        if len(mapped_blocks) == 0:
+            self.logger.debug(f"Tipreport {tipreport_id} already up to date")
+            return
+
+        self.logger.debug(f"Applying patch content blocks to existing tipreport {tipreport_id}, last modified at {last_modified.strftime('%B %d, %Y %I:%M %p')}")
+
+        response = requests.patch(url=patch_url,
+                    json=patch_payload,
+                    headers=self.headers)
+
+        if response.status_code not in SUCCESSFUL_HTTP_CODES:
+            raise HTTPException(f"Cannot patch report, {response.content}, {response.status_code}")
+
+
+        self.logger.debug(f"Patch response: {response.status_code}, {response.content}")
+
+
+
+
+    def add_new_tipreport(self, advisory):
+        add_tipreport_url = f"{self.anomali_url}/api/v1/tipreport/"
+
+        advisory_id = advisory['id']
+        content_blocks = advisory['content_blocks']
+        earliest_content_block = advisory['content_blocks'][-1]
+
+        report_model: AnomaliTipReportModel = AnomaliTipReportModel(
+            body=earliest_content_block['executive_summary'],
+            created_ts=advisory['timestamp_created'],
+            modified_ts=advisory['timestamp_updated'],
+            name=advisory['title'],
+            tags=[
+                self.get_world_watch_tag(advisory_id),
+                self.GENERIC_WORLD_WATCH_BULLETIN_TAG
+            ]
+        )
+
+        response = requests.post(add_tipreport_url, json=report_model.model_dump(), headers=self.headers)
+
+        self.logger.debug("Added new bulletin")
```

Comment:
improve debugging like:
```suggestion
        if response.status_code not in SUCCESSFUL_HTTP_CODES:
            self.logger.error(f"Cannot add new bulletin {advisory['id']} - \"{advisory['title']}\", {response.content}, {response.status_code}")
        else:
            self.logger.info(f"Successfully added new bulletin {advisory['id']} - \"{advisory['title']}\" with Anomali id {response.json()['id']}")
```